src/Clapi/Relay.hs

```diff
@@ -209,3 +225,24 @@ relay vs = waitThen fwd (const $ error "message from the void")
         sendRev $ ServerData awu ms'
         relay vs'
     fwd _ = relay vs
+
+data NsChange =
+    NsAssign Name Path
+  | NsRemove Name
+
+nsChange :: VsDelta -> Maybe NsChange
+nsChange (p, Left tp) = if isNamespace p then Just $ NsAssign (head p) tp else Nothing
```

Comment:
Eeep, `head` - presuming this is safe here because you "know" that your path will never be `root`?

Better might be to define a function (near `isNamespace`) that produced a value on only namespace:
```haskell
maybeNamespace :: Path -> (Name -> Maybe a) -> Maybe a
maybeNamespace (n:[]) f = Just $ f n
maybeNamespace _ = Nothing
```
Then you could write `nsChange` as:
```haskell
nsChange (p, Left tp) = maybeNamespace (flip NsAssign tp) p
nsChange (p, Right Tree.Delete) = maybeNamespace NsRemove p
nsChange _ = Nothing
```